src/screens/authentication/create_new_wallet/RecoveryPhraseScreen/index.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {useEffect, useState} from 'react';
+import {View, StyleSheet, ScrollView} from 'react-native';
+import {CLayout, CHeader} from "components";
+import {colors} from "assets";
+import {scale} from "device";
+import {PhraseItem} from "../components";
+import {Row} from 'components';
+import CButton2 from "components/CButton2";
+import {useNavigation} from "@react-navigation/native";
+import CreateNewWalletRouter from "navigation/CreateNewWalletNavigation/CreateNewWalletRouter";
+import {StackNavigationProp} from "@react-navigation/stack";
+import {Phrase} from "screens/authentication/data/data";
+
+const phraseString = 'House Ego Assits Repair Respond Attitude Different Difficult Opposition Resident Populate Inhabit Situated Problem Failed Name Octupus Doctor Strange Ironman Capital Dimondhand Flash Vision';
+
+const RecoveryPhraseScreen = () => {
+
+    const {navigate} = useNavigation<StackNavigationProp<any>>();
+
+    const [listLeft, setListLeft] = useState<Array<Phrase>>([]);
+    const [listRight, setListRight] = useState<Array<Phrase>>([]);
+    const [data, setData] = useState<Array<Phrase>>([]);
+
+    useEffect(() => {
+        const arrayData = phraseString.split(/\s+/).map((word, index) => {
+            return {
+                id: index + 1,
+                word: word
+            }
+        });
+        if (arrayData && arrayData.length > 0) {
+            setData(arrayData);
+            const left = arrayData.slice(0, Math.round(arrayData.length / 2));
+            const right = arrayData.slice(left.length, arrayData.length);
+
+            setListLeft(left);
+            setListRight(right);
+        }
+    }, []);
+
+    const openDoubleCheckIt = () => {
+        if (data && data.length > 0) {
+            navigate(CreateNewWalletRouter.DOUBLE_CHECK_IT_SCREEN, {data: JSON.parse(JSON.stringify(data))});
```

Comment:
Are there any special reason for stringify and parse data again ? if yes please wrap them with try catch since JSON.parse could throw exception.

Reply:
If you pass normal data, or {data: [...data]}, the selected phrases won't change them state. 
[https://drive.google.com/file/d/1dNsxt7mzqvwl1C7Ad_anLuzE-xYQ1yUs/view?usp=sharing](url)